File: bench/tasks/16-map-filter/go/main.go
package main

import "fmt"

func main() {
	xs := []int{1, 2, 3, 4, 5, 6, 7, 8}
	midCount, highSum := 0, 0
	for _, x := range xs {
		n := x * 3
		if n > 10 && n < 20 {
			midCount++
		}
		if n > 10 {
			highSum += n
		}
	}
	fmt.Println(midCount)
	fmt.Println(highSum)
}

File: bench/tasks/16-map-filter/rust/main.rs
fn main() {
    let xs = [1, 2, 3, 4, 5, 6, 7, 8];
    let mapped: Vec<i32> = xs.iter().map(|x| x * 3).collect();
    let mid_count = mapped.iter().filter(|&&n| n > 10 && n < 20).count();
    let high_sum: i32 = mapped.iter().copied().filter(|&n| n > 10).sum();
    println!("{}", mid_count);
    println!("{}", high_sum);
}
